### ISSUE TITLE
fault accessing __name__ of a native bound method

### ISSUE THREAD

**Issue by majo456** [micropython/micropython#4028]
There is a problem in last version (v1.9.4-429-ge755bd493) of micropython for ESP32. It looks like bad implementation of dir() function. The example below will result in core dump and rebooting the device.

Example:
```
 import network
 ap = network.WLAN(network.AP_IF)
 dir(ap.config)
```

Result:
```
MicroPython v1.9.4-429-ge755bd493 on 2018-08-03; ESP32 module with ESP32
Type "help()" for more information.
>>> import uos
>>> uos.uname()
(sysname='esp32', nodename='esp32', release='1.9.4', version='v1.9.4-429-ge755bd493 on 2018-08-03', machine='ESP32 module with ESP32')
>>> dir(print)
['__class__']
>>> import network
>>> ap = network.WLAN(network.AP_IF)
I (313356) wifi: wifi driver task: 3ffcb834, prio:23, stack:4096, core=0
I (313356) wifi: wifi firmware version: ac8d7b4
I (313356) wifi: config NVS flash: enabled
I (313356) wifi: config nano formating: disabled
I (313366) system_api: Base MAC address is not set, read default base MAC address from BLK0 of EFUSE
I (313376) system_api: Base MAC address is not set, read default base MAC address from BLK0 of EFUSE
I (313396) wifi: Init dynamic tx buffer num: 32
I (313396) wifi: Init data frame dynamic rx buffer num: 64
I (313396) wifi: Init management frame dynamic rx buffer num: 64
I (313406) wifi: Init static rx buffer size: 1600
I (313406) wifi: Init static rx buffer num: 10
I (313406) wifi: Init dynamic rx buffer num: 0
>>> dir(ap.config)
Guru Meditation Error: Core  0 panic'ed (LoadStoreError)
. Exception was unhandled.
Core 0 register dump:
PC      : 0x4017a763  PS      : 0x00060030  A0      : 0x800dfb3f  A1      : 0x3ffb6e70
A2      : 0x400f12ed  A3      : 0x00000000  A4      : 0x00000267  A5      : 0x3ffc30c4
A6      : 0x00000268  A7      : 0x00000169  A8      : 0x400f12ec  A9      : 0x3ffb6e60
A10     : 0x00000003  A11     : 0x00060e23  A12     : 0x00060e20  A13     : 0x3ffe6450
A14     : 0x0000002a  A15     : 0x3ffb6fb0  SAR     : 0x0000001e  EXCCAUSE: 0x00000003
EXCVADDR: 0x400f12ec  LBEG    : 0x4000c46c  LEND    : 0x4000c477  LCOUNT  : 0x00000000

Backtrace: 0x4017a763:0x3ffb6e70 0x400dfb3c:0x3ffb6e90 0x400de1ad:0x3ffb6eb0 0x400dbec9:0x3ffb6ed0 0x400dbf97:0x3ffb6ef0 0x400e7159:0x3ffb6f70 0x400dfa4e:0x3ffb6fa0 0x400dbd59:0x3ffb6fd0 0x400e8f61:0x3ffb6ff0 0x400dfad0:0x3ffb7090 0x400dbd59:0x3ffb70f0 0x400dbd86:0x3ffb7110 0x400f8d8e:0x3ffb7130 0x400f8fcc:0x3ffb71d0 0x400efdbc:0x3ffb7210

Rebooting...
ets Jun  8 2016 00:22:57
```

**Comment by dpgeorge** [comment]
Thanks for the report, I can confirm the bug.  The core issue is not with the esp32 or `dir()`, but rather with accessing `__name__` of a bound, native method.  Eg, the following will crash unix uPy:
```python
1 .to_bytes.__name__
```

**Comment by dpgeorge** [comment]
Should be fixed by 113f00a9ab469a83d1004dac502077af0a8f4847
